### PR TITLE
Allow Skipping Activation

### PR DESCRIPTION
**Pull request by AndrewKahr**
#### Changes

- Allow skipping activation. This is helpful for Mac self-hosted runners that may have multiple builds running simultaneously where the action activating/deactivating the license conflicts between jobs
- Misc code cleanup
- Package updates

#### Related PRs

- game-ci/documentation#452

#### Checklist

- [x] Read the contribution [guide](https://github.com/game-ci/unity-builder/blob/main/CONTRIBUTING.md) and accept the
      [code](https://github.com/game-ci/unity-builder/blob/main/CODE_OF_CONDUCT.md) of conduct
- [x] Docs (If new inputs or outputs have been added or changes to behavior that should be documented. Please make a PR
      in the [documentation repo](https://github.com/game-ci/documentation))
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

File: src/model/build-parameters.ts [hunk 1]
```diff
@@ -190,7 +192,7 @@ class BuildParameters {
       branch: Input.branch.replace('/head', '') || (await GitRepoReader.GetBranch()),
       cloudRunnerBranch: CloudRunnerOptions.cloudRunnerBranch.split('/').reverse()[0],
       cloudRunnerDebug: CloudRunnerOptions.cloudRunnerDebug,
-      githubRepo: Input.githubRepo || (await GitRepoReader.GetRemote()) || 'game-ci/unity-builder',
+      githubRepo: (Input.githubRepo ?? (await GitRepoReader.GetRemote())) || 'game-ci/unity-builder',
```
Comment (webbertakken) [inline src/model/build-parameters.ts:195]: Why the change to nullish coalesce? Should it not be fixed if that would ever throw?

File: src/model/input.ts [hunk 2]
```diff
@@ -186,35 +186,35 @@ class Input {
   }
 
   static get sshPublicKeysDirectoryPath(): string {
-    return Input.getInput('sshPublicKeysDirectoryPath') || '';
+    return Input.getInput('sshPublicKeysDirectoryPath') ?? '';
```
Comment (webbertakken) [inline src/model/input.ts:189]: This hides errors and is a bit of an antipattern.

I really believe we shouldn't be converting all or operators to nullish coalesce.

**Review by webbertakken — APPROVED** [review]
Great work!

File: src/model/build-parameters.ts (hunk 1 above)
Comment (AndrewKahr) [reply src/model/build-parameters.ts:195]: Can certainly revert it. Was just a suggestion from sonarlint that I was accepting throughout the file.

File: src/model/input.ts (hunk 2 above)
Comment (AndrewKahr) [reply src/model/input.ts:189]: Just curious, doesn't the || do the same thing for hiding issues? I thought nullish was just a more robust version of ||.
